### ISSUE TITLE
distill version of quarto

### ISSUE THREAD

**Issue by shoibalc** [quarto-dev/quarto-cli#95]
Any plan to also add to quatro a distill theme that makes it possible to write distill type articles and blogs using formats besides rmarkdown.

**Comment by jjallaire** [comment]
We, we are definitely planning on bringing the distill features into quarto -- stay tuned!

**Comment by mrworthington** [comment]
Curious if there's an update on the distill features? I just tried some of [the examples in here](https://quarto.org/docs/authoring/create-citeable-articles.html#web-articles), which give the impression distill features work, but after knitting the qmd... instead of the author name, it prints "true" and omits the subtitle.

## YAML Screenshot

<img width="844" alt="Screen Shot 2022-03-03 at 9 21 04 PM" src="https://user-images.githubusercontent.com/8171106/156693060-229a0e13-42a6-4599-a8c6-d780d11f70af.png">

## Output Screenshot

<img width="839" alt="Screen Shot 2022-03-03 at 9 20 12 PM" src="https://user-images.githubusercontent.com/8171106/156692982-33590fdb-0785-4103-9e07-11ca111255ea.png">

**Comment by dragonstyle** [comment]
The documentation (on citable documents) got a little ahead of the merged features for the title block. We should had something for you to take a look at later today or this weekend!

**Comment by dragonstyle** [comment]
The latest release should have the handling for title blocks compatible with the YAML you outlined (I made one correction to the quarto.org docs as well). Give it a try and let us know what you think!

**Comment by mrworthington** [comment]
Hey Charles,

First of all, thanks for all y'alls work building this. It's really amazing and I'm interested in switching everything over to quarto from distill. That said, I just tried it. Here's a couple of things I notice as differences from distill right off the bat:

### 1. The metadata from articles don't translate to lists

Take a look at the images below (tagged with quarto v. distill). You can see how the author tag doesn't get listed. Here's what I've got listed for fields on the list page:

```
listing:
  categories: true
  fields: [image, date, description, categories, title, author, reading-time]
```

<img width="1030" alt="Screen Shot 2022-03-04 at 9 05 23 PM" src="https://user-images.githubusercontent.com/8171106/156865419-1cfbb061-0336-494c-ab97-47868a96431f.png">

<img width="1032" alt="Screen Shot 2022-03-04 at 9 01 10 PM" src="https://user-images.githubusercontent.com/8171106/156865367-2e40e9ca-45bd-46e0-b131-b6cfbd4daf83.png">

### 2. The plain text date format doesn't appear to be an option

On [the linked documentation](https://deno.land/std@0.125.0/datetime) for `date_format`, I noticed that a plain text date doesn't appear to be option. If you take a look at both the distill list screenshot above and the distill screenshot below, those formats were possible. But I can't find a way to process that with quarto.

### 3. The article page's title block has less flexibility

This is probably just more on the feedback/feature enhancement side of things, but one of the great things about distill was the flexibility of the title block. With just a small change of css to the background property of the `<d-title>` block , you could make some really amazing aesthetic changes to an article. With quarto, I'm not even certain how you would add an image to the top of the article at all.

<img width="1440" alt="Screen Shot 2022-03-04 at 9 07 34 PM" src="https://user-images.githubusercontent.com/8171106/156865561-a523779c-1aa7-4076-b690-8b635fd8cb80.png">

<img width="1438" alt="Screen Shot 2022-03-04 at 9 08 40 PM" src="https://user-images.githubusercontent.com/8171106/156865564-f5a92052-024c-42e8-9791-d8336e5bbd4e.png">

**Comment by dragonstyle** [comment]
Thank you so much for taking the time to provide suck great feedback! (Great looking article / website, btw!)

> 1. The metadata from articles don't translate to lists

Do you mind sharing the front matter from one of the documents in the listing - definitely shouldn't be doing this! If you're not able to share (no worries!), any clues will help me reproduce the issue!

> 2. The plain text date format doesn't appear to be an option

You're right to note that the date format options are pretty limited right now. Let me take a look at whether we might have some other options to expose a more complete set of formats.

> 3. The article page's title block has less flexibility.

You should still be able to target the title block, though to achieve the 'full bleed' look the you have going is more complicated since the title region is currently within the 'body' of the document, limiting its width. Let me give this some thought, I can definitely think of a couple ways I could make this simple, and it is a great feature.

I definitely also notice in your screenshot that the 'note' text is crowing the title block. Can you tell me where that text is coming from (e.g. is it part of the body of the document) - seems like I need to be sure we have sufficient margin on the bottom of that title block and I'm somehow missing a case.

**Comment by dragonstyle** [comment]
Thanks again for reporting these issue. I have identified and fixed the issues for 1 & 2, and will update this when I've resolved issues with banner style presentation of the title block (3). The latest build should have fixes for issues 1 and 2.

 For `date-format` I've added an option for you to provide a named date style such as `full`, `long`, `medium`, and `short`. I've changed the default behavior to reflect the `medium` style which matches your desired style and looks best for the default cases of listings. I've updated the docs to reflect this new ability here:

https://quarto.org/docs/websites/website-listings.html#advanced-options

**Comment by mrworthington** [comment]
This is great, so thank you very much. Passing along a few notes you requested or I've observed related to the last few comments on this issue:

> I definitely also notice in your screenshot that the 'note' text is crowing the title block. Can you tell me where that text is coming from (e.g. is it part of the body of the document) - seems like I need to be sure we have sufficient margin on the bottom of that title block and I'm somehow missing a case.

Regarding the note, it's just in the body of the .qmd file. Guessing it's a css issue?

<img width="774" alt="Screen Shot 2022-03-06 at 1 44 22 PM" src="https://user-images.githubusercontent.com/8171106/156939664-1f96723a-018f-44a9-b7d9-96bf44f039a3.png">

> I have identified and fixed the issues for 1 & 2

This appears to be working for me. The only thing I noticed here is that the date format only applies to the listed articles, but not the article itself.

<img width="1077" alt="Screen Shot 2022-03-06 at 1 47 28 PM" src="https://user-images.githubusercontent.com/8171106/156939722-30e7cb7f-25cd-4c51-ba5c-e7e9c60f7060.png">

<img width="900" alt="Screen Shot 2022-03-06 at 1 53 10 PM" src="https://user-images.githubusercontent.com/8171106/156939744-6d05a4e0-3075-4074-bfbb-b7b3e9c28a62.png">

> You should still be able to target the title block, though to achieve the 'full bleed' look the you have going is more complicated since the title region is currently within the 'body' of the document, limiting its width. Let me give this some thought, I can definitely think of a couple ways I could make this simple, and it is a great feature.

If it helps, [here's the article](https://www.mrworthington.com/posts/why_americans_no_ed/) with the full-bleed banner. You can inspect the CSS in developer mode on your browser, but I've also pasted it below. It's just adding an image in the background color of that title block.

```
d-title {
padding-top: 12rem;
font-weight: 800;
color: #fff;
font-family: "Gotham SSm A", "Gotham SSm B", -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, "Helvetica Neue", Arial, sans-serif;
background: 
    /* top, transparent red, faked with gradient */ 
    linear-gradient(
      rgba(77,77,77, 0.8), 
      rgba(77,77,77, 0.8)
    ),
    /* bottom, image */
    url(fullsizeoutput_77ff.jpg) no-repeat center center fixed;
-webkit-background-size: cover !important;
-moz-background-size: cover !important;
-o-background-size: cover !important;
background-size:cover !important;
}
```

**Comment by dragonstyle** [comment]
OK, a few more improvements for you in this release https://github.com/quarto-dev/quarto-cli/releases/tag/v0.9.74:

1) The date format is now something you can set on a document as well (though we default to a long style now so depending upon what you're looking for, you may be happy without setting it).

2) You can also specify a banner style header which is much closer to what you're looking for:

```yaml
---
title-block-banner: path-to-banner.jpg
---
```
Docs on the title block here:

https://quarto.org/docs/authoring/title-blocks.html

Love to hear thoughts and reactions... thanks for taking the time to provide such great feedback so far!

**Comment by batpigandme** [comment]
I think we've gotten a disconnect between the headings font weight (i.e. those set using `$headings-font-weight`) and the `h1` in the `title-block-header` now (commenting here because, I, too have largely been porting my distill page content over to Quarto). Easy enough for me to add some custom SCSS/CSS to work around this, but it seems like it'd make sense for the behavior from the current SCSS variable to be inherited. 

One other difference is the behavior with media queries/viewport width (though I'm not 100% sure where this was introduced). The `h1`/page or post title isn't visible rendered as part of the title block when the viewport is narrowed (not sure what the exact breakpoint is—the `.d-none` behavior doesn't seem to affect it `@media (min-width: 1200px)`, but it does _not_ display `@media (max-width: 991.98px)`).

Example (I'm not scrolled differently, the description displays, but the actual title h1 element doesn't)
<img width="2125" alt="Difference in layout between wider and narrower viewports. With the narrower viewport, the title element of the page is no longer part of the display" src="https://user-images.githubusercontent.com/831732/157691988-898f504b-4307-4fce-8e93-bf51e51c6524.png">

**Comment by dragonstyle** [comment]
> I think we've gotten a disconnect between the headings font weight (i.e. those set using `$headings-font-weight`) and the `h1` in the `title-block-header` now (commenting here because, I, too have largely been porting my distill page content over to Quarto). Easy enough for me to add some custom SCSS/CSS to work around this, but it seems like it'd make sense for the behavior from the current SCSS variable to be inherited.

Great catch! Fix will be incoming shortly.

> 
> One other difference is the behavior with media queries/viewport width (though I'm not 100% sure where this was introduced). The `h1`/page or post title isn't visible rendered as part of the title block when the viewport is narrowed (not sure what the exact breakpoint is—the `.d-none` behavior doesn't seem to affect it `@media (min-width: 1200px)`, but it does _not_ display `@media (max-width: 991.98px)`).
> 
> Example (I'm not scrolled differently, the description displays, but the actual title h1 element doesn't) <img alt="Difference in layout between wider and narrower viewports. With the narrower viewport, the title element of the page is no longer part of the display" width="2000" src="https://user-images.githubusercontent.com/831732/157691988-898f504b-4307-4fce-8e93-bf51e51c6524.png">

I think what is happening is that we're expecting the title to 'collapse' into the collapsed sidebar navigation, but there is no sidebar navigation to collapse into. I'm betting when I added the TOC on the right, I probably introduced this hole. Will check and have another fix on the way!

**Comment by dragonstyle** [comment]
https://github.com/quarto-dev/quarto-cli/releases/tag/v0.9.80 should have fixes for both of these issues. Thx for catching them!

**Comment by batpigandme** [comment]
> https://github.com/quarto-dev/quarto-cli/releases/tag/v0.9.80 should have fixes for both of these issues. Thx for catching them!

You're truly the fastest fixer out there!

**Comment by mrworthington** [comment]
Sorry for the _**super**_ long delay. I tried out the updated build and it works. I didn't see the issue Mara did, but great catch! Below is what I was able to build. I know it sounds silly, but this aesthetic plus the floating TOC is pretty fantastic. I think I'm going to port everything over to quarto from distill at this point. I'm sure there's still some kinks to work out, but it's pretty great at this point.

<img width="1437" alt="Screen Shot 2022-03-14 at 2 52 30 PM" src="https://user-images.githubusercontent.com/8171106/158250436-4bdf598a-6b1d-451c-8818-fa786ec3bad1.png">

**Comment by jjallaire** [comment]
I think we're at a point where we can close this issue. Pls feel free to open other issues related to Distill features we haven't implemented yet (not aware of any but I'm sure something has slipped our attention!)

**Comment by batpigandme** [comment]
Yep—everything is solved! @mrworthington, you'd see none of the issues, because @dragonstyle is the quick-draw king of issue fixing, and swatted them in no time flat!

**Comment by mrworthington** [comment]
Howdy y'all!

Reopening this for a quick minute. I just updated quarto to the latest build after not having updated since... not too long after this issue closed. Anyway, was working on a new draft using the title blocks and realized that the `description:` metatag is now misplaced upon publishing with a title block. Before (and in distill), the description tag placed it immediately below the title. After tinkering with it, I realized if you change the `description:` tag to `subtitle:`, it behaves as the `description:` tag ought to. 

The exception to this is that using `subtitle:` doesn't translate to the document listings appropriately. It prints the subtitle and the beginning of the article. If you use `description:`, it just prints what's included in `description:` on the document listings page.

### New Version

<img width="1426" alt="Screen Shot 2022-05-19 at 12 16 24 AM" src="https://user-images.githubusercontent.com/8171106/169218479-3dd55176-e9c2-4c95-89f6-0fe5fcab7eba.png">

### [Previous Version](https://www.mrworthington.com/articles/school_finance/school-finance-analysis/school_finance_viz_us.html)

<img width="1440" alt="Screen Shot 2022-05-19 at 12 30 19 AM" src="https://user-images.githubusercontent.com/8171106/169217329-4c0ea7d6-b515-4e8a-a534-a36510170cee.png">

**Comment by dragonstyle** [comment]
Thanks for pointing this out. We did rewrite the way that we generate the title blocks of sites (to make it easier for you to provide your own template for the title block) so this is something we accidentally broke as a part of that. If you do see others, please let me know as they should be easy to fix and are likely unintentional!

Fixed here 816fce8e8195a5d66878f71526a2ec4f866ba05e and a release with this change should be ready in a few minutes.

**Comment by mrworthington** [comment]
Thank you! I'll definitely call out stuff as I see it.
